dask/core.py

```diff
@@ -194,3 +195,31 @@ def reverse_dict(d):
             result[val].add(k)
     return result
 
+
+def cull(dsk, keys):
+    """ Return new dask with only the tasks required to calculate keys.
+
+    In other words, remove unnecessary tasks from dask.
+    ``keys`` may be a single key or list of keys.
+
+    Exmaple
+    -------
+
+    >>> d = {'x': 1, 'y': (inc, 'x'), 'out': (add, 'x', 10)}
+    >>> cull(d, 'out')  # doctest: +SKIP
+    {'x': 1, 'out': (add, 'x', 10)}
+    """
+    if not isinstance(keys, list):
```

Comment:
Correct.  `tuple` is already overloaded to possibly mean a key or a task.  We use lists as collections because they couldn't possibly mean a single key and so are unambiguous.  Probably I should mention these things more clearly in [the docs](http://dask.readthedocs.org/en/latest/spec.html)